### ISSUE TITLE
Requesting Hebrew or Indonesian instructions falls back to English

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-navigation-android#886]
This library is localized into Hebrew and the Directions API can return instructions in Hebrew. However, if the user sets the locale to Hebrew, they end up with instructions in English. This is because Android’s `Locale.getLanguage()` method uses a deprecated ISO language code for Hebrew.

mapbox/locale-utils#4 tracks a workaround on the server side to accept this deprecated language code. Ideally, this library and the Mapbox Java SDK would both use `Locale.toLanguageTag()` extensively instead of `getLanguage()`, but `toLanguageTag()` isn’t supported by all the API levels we support. That said, a special case for these three deprecated codes might be worth considering.

A fix for this issue either on the server side or on the client side will expose a bug preventing this SDK from falling back to the Android TTS engine when the Voice API lacks support for an instruction’s language: #882. Since the Voice API lacks support for Hebrew, it’ll attempt to read the Hebrew text using an English voice. Amazon Polly’s English voice inserts conspicuous pauses for any writing system other than Latin, so the user could hear road names and route numbers interspersed by pregnant pauses.

This issue also affects Indonesian and Yiddish (the latter of which isn’t supported by either this library or the Directions API). As part of this issue, we should also check whether [`values-id`](https://github.com/mapbox/mapbox-navigation-android/blob/master/app/src/main/res/values-id/strings.xml) and [`values-he`](https://github.com/mapbox/mapbox-navigation-android/blob/master/app/src/main/res/values-he/strings.xml) need to be renamed to `values-in` and `values-iw`, respectively.

To recap:

* [x] Implement fallbacks for deprecated language codes in the Directions API: mapbox/locale-utils#4
* [ ] Possibly backstop usage of `Locale.getLanguage()` in this library and the Mapbox Java SDK
* [ ] Possibly rename values-id and values-he in this library

/cc @danesfeder @devotaaabel @danpaz @Guardiola31337 @bsudekum

**Comment by 1ec5** [comment]
> As part of this issue, we should also check whether [`values-id`](https://github.com/mapbox/mapbox-navigation-android/blob/master/app/src/main/res/values-id/strings.xml) and [`values-he`](https://github.com/mapbox/mapbox-navigation-android/blob/master/app/src/main/res/values-he/strings.xml) need to be renamed to `values-in` and `values-iw`, respectively.

Transifex always uses standard IETF language tags (essentially ISO 639 codes plus any necessary script and region codes). So it’s up to this line to map from the standard language tags to whatever Android expects for the folder names:

https://github.com/mapbox/mapbox-navigation-android/blob/8e81c16035a3cd1fe0b36dfe5c34b8bcebd4e684/.tx/config#L4

**Comment by akitchen** [comment]
Unblocked by https://github.com/mapbox/mapbox-navigation-android/issues/924 and https://github.com/mapbox/locale-utils/issues/4
